examples/model_evaluation/plot_feature_importance.py

```diff
@@ -0,0 +1,492 @@
+
+TableReport(X_y)
+
+# %%
+# From the table report, we can draw some key observations:
+#
+# - Looking at the *Stats* tab, all features are numerical and there are no
+#   missing values.
+# - Looking at the *Distributions* tab, we can notice that some features have some
+#   outliers: ``MedInc``, ``AveRooms``, ``AveBedrms``, ``Population``, and ``AveOccup``.
+#   The feature with the largest number of outliers is ``AveBedrms``, probably
+#   corresponding to vacation resorts.
+# - Looking at the *Associations* tab, we observe that the target feature
+#   ``MedHouseVal`` is mostly associated with ``MedInc``, ``Longitude``, and
+#   ``Latitude``, which makes sense.
+
+# %%
+# Target feature
+# --------------
+#
+# Moreover, the target distribution has a long tail:
+
+# %%
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+sns.histplot(data=X_y, x=target_name, bins=100)
+plt.show()
+
+# %%
+# There seems to be a threshold-effect for high-valued houses: all houses with a price
+# above $500,000 are given the value $500,000.
+
+# %%
+
+# %%
+# Now, let us assess how the median income ``MedInc`` relates to the median house
+# prices ``MedHouseVal``:
+
+# %%
+import plotly.express as px
+
+X_y_plot = X_y.copy()
+X_y_plot["MedInc_bins"] = pd.qcut(X_y_plot["MedInc"], q=5)
+bin_order = X_y_plot["MedInc_bins"].cat.categories.sort_values()
+fig = px.histogram(
+    X_y_plot,
+    x=target_name,
+    color="MedInc_bins",
+    category_orders={"MedInc_bins": bin_order},
+)
+fig
+
+# %%
+# As could have been expected, a high salary often comes with a more expensive house.
+# We can also notice the clipping effect of house prices for very high salaries.
+
+# %%
+# Geospatial features
+# -------------------
+
+# %%
+# Let us look into the coordinates of the districts in California, with regards to the
+# target feature:
+
+
+# %%
+def plot_map(df, color_feature):
+    fig = px.scatter_mapbox(
+        df, lat="Latitude", lon="Longitude", color=color_feature, zoom=5, height=600
+    )
+    fig.update_layout(
+        mapbox_style="open-street-map",
+        mapbox_center={"lat": df["Latitude"].mean(), "lon": df["Longitude"].mean()},
+        margin={"r": 0, "t": 0, "l": 0, "b": 0},
+    )
+    return fig
+
+
+# %%
+fig = plot_map(X_y, target_name)
+fig
+
+# %%
+# As could be expected, the price of the houses near the sea is higher, especially
+# around big cities like Los Angeles and San Jose.
+# Taking into account the coordinates in our modelling will be very important.
+
+# %%
+# Splitting the data
+# -----------------
+
+# %%
+# Just before diving into our first model, let us split our data into a train and a
+# test split:
+
+# %%
+from sklearn.model_selection import train_test_split
+
+X = X_y.drop(columns=[target_name])
+y = X_y[target_name]
+X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
+
+# %%
+# Linear models
+# =============
+
+# %%
+# For our regression task, let us start with linear models.
```

Comment:
A bit redundant with line 10: https://github.com/probabl-ai/skore/blob/a577cad91663adfb61dc9b66e7fb220aac97339a/examples/model_evaluation/plot_feature_importance.py#L10? And with line 183 too https://github.com/probabl-ai/skore/blob/a577cad91663adfb61dc9b66e7fb220aac97339a/examples/model_evaluation/plot_feature_importance.py#L183?